File: src/Neuralm.Services/Neuralm.Services.TrainingRoomService/Neuralm.Services.TrainingRoomService.Messages/GetOrganismsRequest.cs
using System;
using Neuralm.Services.Common.Messages;
using Neuralm.Services.Common.Messages.Abstractions;

namespace Neuralm.Services.TrainingRoomService.Messages
{
    /// <summary>
    /// Represents the <see cref="GetOrganismsRequest"/> class.
    /// </summary>
    [Message("Get", "organisms", typeof(GetOrganismsResponse))]
    public class GetOrganismsRequest: Request
    {
        /// <summary>
        /// Gets and sets the training session id.
        /// </summary>
        public Guid TrainingSessionId { get; set; }
        
        /// <summary>
        /// Gets and sets the amount of organisms to be requested.
        /// </summary>
        public int Amount { get; set; }
        
        /// <summary>
        /// Gets and sets the user id.
        /// </summary>
        public Guid UserId { get; set; }
    }
}

File: src/Neuralm.Presentation.Web/src/messages/requests/GetOrganismsRequest.ts
import Request from './Request';

/**
 * Represents the get organisms request class.
 */
export default class GetOrganismsRequest extends Request {
  public userId: string;
  public trainingSessionId: string;
  public amount: number;

  /**
   * Initializes the GetOrganismsRequest class.
   */
  constructor(userId: string, trainingSessionId: string, amount: number) {
    super();
    this.userId = userId;
    this.trainingSessionId = trainingSessionId;
    this.amount = amount;
  }
}
